hw8/task1.py

```diff
@@ -0,0 +1,149 @@
+"""
+We have a file that works as key-value storage, each like is represented as key and value separated by = symbol, example:
+
+name=kek last_name=top song_name=shadilay power=9001
+
+Values can be strings or integer numbers. If a value can be treated both as a number and a string,
+ it is treated as number.
+
+Write a wrapper class for this key value storage that works like this:
+
+storage = KeyValueStorage('path_to_file.txt')
+that has its keys and values accessible as collection items and as attributes.
+Example:
+    storage['name'] # will be string 'kek'
+    storage.song_name # will be 'shadilay'
+    storage.power # will be integer 9001
+
+In case of attribute clash existing built-in attributes take precedence.
+In case when value cannot be assigned to an attribute (for example when there's a line 1=something)
+ ValueError should be raised.
+ File size is expected to be small, you are permitted to read it entirely into memory.
+"""
+from os import path
+from typing import Any
+
+
+def change_or_delete_line_from_file(
+    filepath: str, key: str, value: Any, delete: bool = False
+):
+    if not path.isfile(filepath):
+        raise ValueError
+
+    result_file_str = ""
+    with open(filepath, mode="r") as f:
+        for line in f:
+            file_key, old_value = line.strip().split("=")
+            if file_key == key:
+                if delete:
+                    continue
+                result_file_str += f"{file_key}={value}\n"
+            else:
+                result_file_str += f"{file_key}={old_value}\n"
+    with open(filepath, mode="w") as f:
+        f.write(result_file_str)
+
+
+class KeyValueStorage:
+    def __init__(self, file_path: str):
+        super().__setattr__(
```

Comment:
Done

Reply:
if it's about
```
if name in self.__dict__:
            super().__setattr__(name, value)
            return
``` 
than it can be simplified to 
```
if hasattr(self, name):
   ...
```